go/types/path.go

```diff
@@ -65,29 +65,12 @@ func constructPath(p Path, str string) (Path, error) {
 			return Path{}, errors.New("[ is missing closing ]")
 		}
 		rem = rem[1:]
-
-		// TODO: Is there some prettier way to collapse this w/ the common case for annotations
-		// below?
-		intoKey := false
-		if len(rem) > 1 && rem[0] == '@' {
-			if ann, rem2 := getAnnotation(rem[1:]); ann != "" {
-				if ann == "key" {
-					intoKey = true
-					rem = rem2
-				}
-			}
-		}
-
 		d.Chk.NotEqual(idx == nil, h.IsEmpty())
 
 		var part PathPart
 		switch {
```

Comment:
`switch` here is overkill now, it can be much simpler:
```go
if idx != nil {
  p = append(p, NewIndexPath(idx))
} else {
  p = append(p, NewHashIndexPath(h))
}
```